src/bp/nlu/stan/http-typings.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  TrainingProgress,
+  PredictOutput,
+  IntentDefinition,
+  EntityDefinition,
+  Specifications,
+  Health
+} from 'nlu/typings_v1'
+
+export interface Credentials {
+  appId: string
+  appSecret: string
+}
+
+export interface TrainRequestBody extends Credentials {
+  language: string
+  contexts: string[]
+  intents: IntentDefinition[]
+  entities: EntityDefinition[]
+  seed?: number
+}
+
+export interface PredictRequestBody extends Credentials {
+  utterances: string[]
+}
+
+export interface DetectLangRequestBody extends PredictRequestBody {
+  models: string[]
+}
+
+export interface ErrorResponse {
+  success: false
+  error: string
```

Comment:
Shouldn't an http error have at least an error code and a message field?

Reply:
Well the error code is not part of the Response body, but you're right; it can be added. Also, for now the `error` field is the whole message. I plan on adding the stack trace and other things, but for now I only wanted to type the current code.